kukur/auth.py

```diff
@@ -0,0 +1,51 @@
+"""Kukur module for authentication and authorization."""
+
+import time
+from dataclasses import dataclass
+
+import requests
+from requests.auth import AuthBase
+
+
+@dataclass
+class OIDCConfig:
+    """Configuration for OIDC authentication."""
+
+    client_id: str
+    client_secret: str
+    oidc_token_url: str
+
+
+class OIDCBearerAuth(AuthBase):
+    """Handle OIDC bearer token authorization.
+
+    Token is refreshed when expired.
+    """
+
+    def __init__(self, config: OIDCConfig):
+        self.token_url = config.oidc_token_url
+        self.client_id = config.client_id
+        self.client_secret = config.client_secret
+        self._access_token = None
+        self._expires_at = 0
+
+    def _refresh_token(self):
```

Comment:
```suggestion
    def _refresh_token(self) -> None:
```

Otherwise this is not typechecked.